lib/component.js

```diff
@@ -172,6 +172,16 @@
     // Use `nextProps` or `component.props` and grab `model` and `collection`
     // from there
     var props = nextProps || component.props || {};
+    if (component.overrideModel && typeof component.overrideModel === 'function'){
```

Comment:
I think overriding is better since we have access to `props.model` in the override method and we can return that if we need. Otherwise we can return anything else.